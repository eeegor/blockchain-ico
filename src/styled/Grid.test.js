import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import {
	Grid,
	makeMobileColumns,
	makeTabletColumns,
	makeDesktopColumns
} from './Grid';

describe('Grid', () => {
	test('calculates default mobile columns', () => {
		const columns = makeMobileColumns(3);
		expect(columns).toEqual(1);
	});

	test('calculates responsive mobile columns', () => {
		const columns = makeMobileColumns([3, 5]);
		expect(columns).toEqual(3);
	});

	test('calculates default tablet columns', () => {
		const columns = makeTabletColumns(3);
		expect(columns).toEqual(3);
	});

	test('calculates responsive tablet columns', () => {
		const columns = makeTabletColumns([3, 5]);
		expect(columns).toEqual(5);
	});

	test('calculates default desktop columns', () => {
		const columns = makeDesktopColumns(3);
		expect(columns).toEqual(3);
	});

	test('calculates responsive desktop columns', () => {
		const columns = makeDesktopColumns([3, 5, 8]);
		expect(columns).toEqual(8);
	});

	test('renders default grid', () => {
		const tree = renderer.create(<Grid>Content</Grid>).toJSON();
		expect(tree).toMatchSnapshot();
	});

	test('renders grid with given number of columns', () => {
		const tree = renderer.create(<Grid top={4}>Content</Grid>).toJSON();
		expect(tree).toMatchSnapshot();
	});

	test('renders grid with given number of responsive columns [mobile, tablet, desktop]', () => {
		const tree = renderer
			.create(<Grid top={[2, 3, 4]}>Content</Grid>)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
