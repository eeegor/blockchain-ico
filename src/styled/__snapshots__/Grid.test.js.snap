// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid renders default grid 1`] = `
.c0 {
  display: grid;
  grid-gap: 15px 15px;
  grid-template-columns: repeat( 1,1fr );
}

@media (min-width:767px) {
  .c0 {
    grid-gap: 30px 30px;
    grid-template-columns: repeat( 1,1fr );
  }
}

@media (min-width:1024px) {
  .c0 {
    grid-template-columns: repeat( 1,1fr );
  }
}

<div
  className="c0"
>
  Content
</div>
`;

exports[`Grid renders grid with custom gap 1`] = `
.c0 {
  display: grid;
  grid-gap: 15px 15px;
  grid-template-columns: repeat( 1,1fr );
  grid-gap: 6px 6px;
}

@media (min-width:767px) {
  .c0 {
    grid-gap: 6px 6px;
    grid-template-columns: repeat( 1,1fr );
  }
}

@media (min-width:1024px) {
  .c0 {
    grid-template-columns: repeat( 1,1fr );
  }
}

<div
  className="c0"
>
  Content
</div>
`;

exports[`Grid renders grid with given number of columns 1`] = `
.c0 {
  display: grid;
  grid-gap: 15px 15px;
  grid-template-columns: repeat( 1,1fr );
}

@media (min-width:767px) {
  .c0 {
    grid-gap: 30px 30px;
    grid-template-columns: repeat( 1,1fr );
  }
}

@media (min-width:1024px) {
  .c0 {
    grid-template-columns: repeat( 1,1fr );
  }
}

<div
  className="c0"
>
  Content
</div>
`;

exports[`Grid renders grid with given number of responsive columns [mobile, tablet, desktop] 1`] = `
.c0 {
  display: grid;
  grid-gap: 15px 15px;
  grid-template-columns: repeat( 1,1fr );
}

@media (min-width:767px) {
  .c0 {
    grid-gap: 30px 30px;
    grid-template-columns: repeat( 1,1fr );
  }
}

@media (min-width:1024px) {
  .c0 {
    grid-template-columns: repeat( 1,1fr );
  }
}

<div
  className="c0"
>
  Content
</div>
`;
